lib/wab/impl/data.rb

```diff
@@ -83,73 +71,51 @@ def set(path, value, repair=false)
         else
           validate_value(value)
         end
-        node = @root
         path = path.to_s.split('.') unless path.is_a?(Array)
-        path[0..-2].each { |key|
-          if node.is_a?(Hash)
-            key = key.to_sym
-            node[key] = {} unless node.has_key?(key)
-            node = node[key]
-          elsif node.is_a?(Array)
-            key = key_to_int(key)
-            if key < node.length && -node.length < key
-              node = node[key]
-            else
-              nn = {}
-              if key < -node.length
-                node.unshift(nn)
-              else
-                node[key] = nn
-              end
-              node = nn
-            end
-          else
-            raise StandardError.new("Can not set a member of an #{node.class}.")
-          end
-        }
+        node = assign_node(path, root)
+
         key = path[-1]
+        ensure_hash_or_array(node)
         if node.is_a?(Hash)
           key = key.to_sym
           node[key] = value
-        elsif node.is_a?(Array)
+        else # node is an instance of Array
           key = key_to_int(key)
           if key < -node.length
             node.unshift(value)
           else
             node[key] = value
           end
-        else
-          raise StandardError.new("Can not set a member of an #{node.class}.")
         end
         value
       end
 
       # Each child of the Data instance is provided as an argument to a block
       # when the each method is called.
       def each(&block)
-        each_node([], @root, block)
+        each_node([], root, block)
       end
 
       # Each leaf of the Data instance is provided as an argument to a block
       # when the each method is called. A leaf is a primitive that has no
       # children and will be nil, a Boolean, String, Numberic, Time, WAB::UUID,
       # or URI.
       def each_leaf(&block)
-        each_leaf_node([], @root, block)
+        each_leaf_node([], root, block)
       end
 
       # Make a deep copy of the Data instance.
       def clone()
         # avoid validation by using a empty Hash for the intial value.
         c = self.class.new({}, false)
-        c.instance_variable_set(:@root, clone_value(@root))
+        c.instance_variable_set(:@root, clone_value(root))
         c
       end
 
       # Returns the instance converted to native Ruby values such as a Hash,
       # Array, etc.
       def native()
-        @root
+        root
```

Comment:
Using an alias instead of calling the root method would have to be faster just since it reduces one call. @root vs root timing I'm not sure of. There might be some optimization somewhere for that but at least in older versions of Ruby @root would have been faster.

Another plus for using an alias is that it is just one line. Not that important though.